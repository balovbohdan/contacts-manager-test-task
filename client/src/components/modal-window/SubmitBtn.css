.submitBtn {
    cursor: pointer;
    padding: 10px 15px;
    transition: var(--trans);
    border-radius: var(--border-radius);
    background-color: rgba(0, 0, 0, .2);
}

.submitBtn:hover {
    background-color: rgba(0, 0, 0, .3);
}

.submitBtn__disabled {
    cursor: not-allowed;
}

.submitBtn__disabled:hover {
    background-color: rgba(0, 0, 0, .2);
}