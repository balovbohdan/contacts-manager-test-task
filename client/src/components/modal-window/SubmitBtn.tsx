import * as React from 'react';
import classnames from 'classnames';

import * as submitBtnCss from '@css/submit-btn.css';

type Props = {
    submit:()=>void;
    isDataValid:boolean;
};

export const SubmitBtn = ({submit, isDataValid}:Props) => {
    const doSubmit = () => isDataValid && submit();

    const className = classnames({
        [submitBtnCss.submitBtn]: true,
        [submitBtnCss.submitBtn__disabled]: !isDataValid
    });

    return <div onClick={doSubmit} className={className}>Submit</div>;
};