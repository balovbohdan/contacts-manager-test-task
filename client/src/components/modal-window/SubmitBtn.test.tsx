import * as React from 'react';
import {shallow} from 'enzyme';

import {SubmitBtn} from './SubmitBtn';

describe('Submit modal window btn', () => {
    it('Should match snapshot', () => {
        const props = {
            submit: jest.fn(),
            isDataValid: true
        };

        const btn = shallow(<SubmitBtn {...props}/>);

        expect(btn).toMatchSnapshot();
    });

    it('Should react on click if data is valid', () => {
        const props = {
            submit: jest.fn(),
            isDataValid: true
        };

        const btn = shallow(<SubmitBtn {...props}/>);

        btn.simulate('click');
        expect(props.submit).toBeCalledTimes(1);
    });

    it('Should not react on click if data is invalid', () => {
        const props = {
            submit: jest.fn(),
            isDataValid: false
        };

        const btn = shallow(<SubmitBtn {...props}/>);

        btn.simulate('click');
        expect(props.submit).toBeCalledTimes(0);
    });
});