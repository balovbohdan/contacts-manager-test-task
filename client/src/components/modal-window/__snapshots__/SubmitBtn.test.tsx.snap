// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Submit modal window btn Should match snapshot 1`] = `
<div
  className=""
  onClick={[Function]}
>
  Submit
</div>
`;
